tests/test_sso.py

```diff
@@ -108,6 +124,48 @@ def test_acs_sets_session_correctly(self):
             self.assertEqual(samlNameId, sess.get('samlNameId'))
             self.assertEqual(samlSessionIndex, sess.get('samlSessionIndex'))
 
+        self.assertEqual(
+            self.get_sessions(),
+            [
+                {
+                    'id': 1,
+                    'expiry': '2018-10-18T13:30:00',
+                    'data': {
+                        '_permanent': True,
+                        'loggedIn': True,
+                        'samlAttributes': {
+                            'urn:oid:0.9.2342.19200300.100.1.1': ['bruce'],
+                            'urn:oid:0.9.2342.19200300.100.1.3': ['bruce@kung.fu'],
+                            'urn:oid:2.5.4.41': ['Bruce Lee'],
+                        },
+                        'samlNameId': '_e3dfaf3e3385fd182b1c4d4164644393cce3ac7bfe',
+                        'samlSessionIndex': '_6a54ed11e21b64af1a0380b1fba3ec575b05855465',
+                        'samlSessionType': flask_saml_sso.session.SessionType.User,
+                    },
+                }
+            ],
+        )
+
+    @freezegun.freeze_time('2018-10-17')
+    def test_purge_retains_current_sessions(self):
+        self.test_acs_sets_session_correctly()
```

Comment:
Having one test call another seems iffy. Refactor `test_acs_sets_session_correctly` and split out the part you want to reuse.